File: frontend/src/types/summary.ts
export interface FileSummary {
  id: number;
  file_id: number;
  executive_summary: string;
  key_points: string[];
  important_dates: string[];
  people: string[];
  organizations: string[];
  risks: string[];
  action_items: string[];
  model: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/summary.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class FileSummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    file_id: int
    executive_summary: str
    key_points: list[str]
    important_dates: list[str]
    people: list[str]
    organizations: list[str]
    risks: list[str]
    action_items: list[str]
    model: str
    created_at: datetime
    updated_at: datetime
